spec.yaml

```diff
@@ -97,7 +97,7 @@ paths:
           },
           "workers": {
             "count_per_cluster": {
-              "max": 20,
+              "max": -1,
               "default": 3
             },
```

Comment:
So the suggestion in the PM was to define the count as `null` for cloud providers. There are multiple values and the `default` is actually useful still IMO. Making a numeric value `null` does not sound quite right to me. So how do we want to express "unlimited" resources? I made it here `-1` as a suggestion. WDYT?

Reply:
@xh3b4sd let's maybe go with 500 then. We haven't had clusters that big and probably nobody should create clusters that big. We can adjust this again once we have more information which size of clusters is problematic for our management.

Using `-1` is weird imo.

Reply:
The immediate urge for the PM got already removed by setting it to `100`. It is just that these static numbers are weird and wrong. We should actually have a way to express no limit or indicate the number is unknown. In case you pay for the trusted advisor service you can get some limits I guess and we could add such a support in case the service is available, but it would be nice from the UX point of view to have a mechanism to express that a number is unknown anyway.

Reply:
`null` was specifically to indicate that this number is unknown. I think within computing it's pretty widely understood to mean undefined or unknown, which unlimited also is, because resources always have a limit, it's just that it's not known what breaks first when there's high enough number of workers (most probably master runs out of resources at that point, but could be also IP address space or something else).

If we implement check with TrustedAdvisor, we should probably then also account for other limits (such as mentioned subnet size and master instance type).

Reply:
Eh but I would like to have some type consistency somehow and `null` implies that you have to deal with that language specifically and everything is just more complicated. In golang we would then need to define an integer pointer or something like that in order to get `nil` and definitely know about `null` being set. Otherwise the result is `0` anyway after unmarshalling and we did not gain anything.

Reply:
See https://play.golang.org/.

Reply:
Sure, dealing with pointers is slightly cumbersome in this case, but in external interfaces, especially APIs, the clarity and semantic consistency has a lot bigger priority (in my point of view) than code that has to deal with it. This is language specific (and actually library specific, see e.g. how protobuf generates code or how aws-sdk is implemented) implementation detail and shouldn't affect API contract.